File: lexer/ParserError.cs
// ============================================================
// ParserError.cs
// Diagnóstico (error) producido durante el análisis sintáctico.
// ============================================================

namespace BasicPlus.Lexer;

public sealed class ParserError
{
    public string Message { get; }
    public int Line { get; }
    public int Column { get; }

    public ParserError(string message, int line, int column)
    {
        Message = message;
        Line = line;
        Column = column;
    }

    public override string ToString() => $"[{Line}:{Column}] error sintáctico: {Message}";
}

File: lexer-java/src/main/java/basicplus/frontend/ParserError.java
// ============================================================
// ParserError.java
// Diagnóstico (error) producido durante el análisis sintáctico.
// ============================================================
package basicplus.frontend;

public final class ParserError {
    public final String message;
    public final int line;
    public final int column;

    public ParserError(String message, int line, int column) {
        this.message = message;
        this.line = line;
        this.column = column;
    }

    @Override
    public String toString() {
        return "[" + line + ":" + column + "] error sintáctico: " + message;
    }
}
